db/migrate/20241103224635_create_user_roles.rb

```diff
@@ -0,0 +1,28 @@
+class CreateUserRoles < ActiveRecord::Migration[7.1]
+  def up
+    create_table :roles do |t|
+      t.string :name
+
+      t.timestamps
+    end
+
+    create_table :user_roles do |t|
+      t.belongs_to :user
+      t.belongs_to :role
+
+      t.timestamps
+    end
```

Comment:
Any reason we decided to put these in the same migration?
Instead of separate migrations?

Reply:
It as another decision based on convenience: the change is simple, part of the same feature and in an error scenario, rolling the migration back won't affect the rest of the existing schema.